lightning/src/ln/outbound_payment.rs

```diff
@@ -747,11 +747,11 @@ impl OutboundPayments {
 	{
 		match err {
 			PaymentSendFailure::AllFailedResendSafe(errs) => {
-				Self::push_payment_path_failed_evs(payment_id, payment_hash, route.paths, errs.into_iter().map(|e| Err(e)), pending_events);
+				Self::push_payment_path_failed_evs(payment_id, payment_hash, Some(&mut route_params), route.paths, errs.into_iter().map(|e| Err(e)), pending_events);
 				self.retry_payment_internal(payment_id, route_params, router, first_hops, inflight_htlcs, entropy_source, node_signer, best_block_height, logger, pending_events, send_payment_along_path);
 			},
 			PaymentSendFailure::PartialFailure { failed_paths_retry: Some(retry), results, .. } => {
-				Self::push_payment_path_failed_evs(payment_id, payment_hash, route.paths, results.into_iter(), pending_events);
+				Self::push_payment_path_failed_evs(payment_id, payment_hash, None, route.paths, results.into_iter(), pending_events);
```

Comment:
Why `None` here?

Reply:
We could push the failed scids here, but they won't be used because we just use the `PartialFailure::failed_paths_retry` for resend, which already contains them.

Reply:
> when looking at fail_htlc for a non-immediate failure, it looks to be added twice. Once when calling insert_previously_failed_scid and another time when determining the failure was not a probe.

So the second add will go away when we get rid of `PaymentPathFailed::retry`, which [as you pointed out](https://github.com/lightningdevkit/rust-lightning/pull/2008#discussion_r1103690969) should be removed. 

> But shouldn't the event that we surface also reflect this? Why would we exclude the scid here but include it for the AllFailedResendSafe case?

We never include failed scids in the event surfaced in `handle_pay_route_err`, `PaymentPathFailed::retry` is always `None`, if I understand you correctly. The scids are being pushed into the route params for immediate retry purposes only.

On a related note, on second glance my statement that we push failed scids into `failed_paths_retry` on `PartialFailure` was not correct. I think I thought that because the issue says it: https://github.com/lightningdevkit/rust-lightning/issues/1969 but it isn't in the code. Fixing this.

Reply:
> We never include failed scids in the event surfaced in `handle_pay_route_err`, `PaymentPathFailed::retry` is always `None`, if I understand you correctly. The scids are being pushed into the route params for immediate retry purposes only.

Ah, right, I missed that. Was thinking it was added from the parameter.